lighthouse-core/lib/network-recorder.js

```diff
@@ -26,10 +27,36 @@ class NetworkRecorder extends EventEmitter {
     this._records = recordArray;
     this.networkManager = NetworkManager.createWithFakeTarget();
 
-    // TODO(bckenny): loadingFailed calls are not recorded in REQUEST_FINISHED.
+    this.startedRequestCount = 0;
+    this.finishedRequestCount = 0;
+    this.idleState = true;
```

Comment:
they are different to keep track of which event we've actually emitted, but good point, we can use the actual number of active requests to derive that state